### ISSUE TITLE
nanopolish variants gives empty output

### ISSUE THREAD

**Issue by scottgigante** [jts/nanopolish#162]
I am running `nanopolish variants` on 110,000 reads of an E.coli dcm-/dam- mutant aligned to the K12 MG1655 genome using the command recommended in the README. Strangely, I am receiving no output whatsoever.

```
% python nanopolish/scripts/nanopolish_makerange.py data/ecoli_k12.fasta | \
    parallel --results nanopolish.results -P 8 nanopolish variants --consensus \
    polished.{1}.fa -w {1} -r data/reads.fasta -b data/reads.sorted.bam -g \
    data/ecoli_k12.fasta -t 2 --min-candidate-frequency 0.1
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  sample
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  sample
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  sample
[...]
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  sample
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  sample
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  sample
```

When I compared the polished sequence to the original genome with 
`cmp <(tail -n +2 data/ecoli_k12.fasta | tr -d '\n') <(tail -n +2 polished_genome.fa | tr -d '\n')`
no differences were found. The output under nanopolish.results is equivalently bare: stderr is empty, and stdout for each range contains only the vcf header.

The same thing also happens without `--consensus` with the following command:

```
% python nanopolish/scripts/nanopolish_makerange.py data/ecoli_k12.fasta | \
    parallel --results nanopolish.results -P 8 nanopolish variants  -w {1} \
    -r data/reads.fasta -b data/reads.sorted.bam -g data/ecoli_k12.fasta -t 2 \
    --min-candidate-frequency 0.1 --ploidy 1
```

I am running nanopolish 0.6.1, updated recently enough that it correctly parses r9.5 reads.

**Comment by jts** [comment]
Do you get any warnings at the end of the run about unparseable reads?

**Comment by scottgigante** [comment]
No warnings at all. I removed the `--results` option from `parallel` and there's nothing at all printed to stderr.

**Comment by jts** [comment]
Hmm, that is odd. Try adding some verbose flags (-vvvvv) and see if
anything informative comes out?

**Comment by scottgigante** [comment]
I tried running with the entire chromosome as a single region to reduce log spam. That's been running on 46 threads (reliably CPU-bound throughout most of the run) for about 24 hours. Log so far looks like this:
```
% nanopolish variants --consensus polished_genome.fa -r data/reads.fasta -b data/reads.sorted.bam -g data/ecoli_k12.fasta -t 46 --min-candidate-frequency 0.1 -vvvvv
input region: AGCTTTTCATTCTGACTGCAACGGGCAAT
[all of E.coli genome]
AATATCACCAAATAAAAAACGCCTTAGTAAGTATTTTTC
Round 1
==== Starting variant screening =====
```

Since that was taking so much longer than previous runs with parallel (and roughly the same number of threads), I tried again with parallel and verbose to see what was causing the problem. With parallel, it ran to completion in 2 minutes and 20 seconds. No useful output.
```
% python ~/nanopolish_makerange.py data/ecoli_k12.fasta | parallel -P 16 nanopolish variants --consensus polished.{1}.fa -w {1} -r data/reads.fasta -b data/reads.sorted.bam -g data/ecoli_k12.fasta -t 46 --min-candidate-frequency 0.1 -vvvvv
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  sample
input region: [...]
Round 1
==== Starting variant screening =====
Round 2
==== Starting variant screening =====
Round 3
==== Starting variant screening =====
Round 4
==== Starting variant screening =====
Round 5
[...]
==== Starting variant screening =====
Round 48
==== Starting variant screening =====
Round 49
==== Starting variant screening =====
Round 50
==== Starting variant screening =====
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  sample
input region: [...]
[...]
Round 48
==== Starting variant screening =====
Round 49
==== Starting variant screening =====
Round 50
==== Starting variant screening =====
```

**Comment by jts** [comment]
It is odd that the 16x46 parallel run would complete so quickly when the full genome run (1x46) takes so long. I'm not sure why that would be. I suggest trying a single 100kb window with 46threads and increase verbosity up to the maximum (`-vvvvvvvvv`). It should print out a score for every base in that 100kb region.

**Comment by scottgigante** [comment]
The single run on 100kb takes proportionate time to the full genome run. Not sure what's going on with the parallel run. However, even the full verbosity leaves me none the wiser.

```
Round 1
==== Starting variant screening =====
NC_000913.3     2000000 .       AT      A       0.0     PASS            (null)  (null)
NC_000913.3     2000001 .       T       A       0.0     PASS            (null)  (null)
NC_000913.3     2000001 .       T       C       0.0     PASS            (null)  (null)
NC_000913.3     2000001 .       T       G       0.0     PASS            (null)  (null)
NC_000913.3     2000001 .       T       TA      0.0     PASS            (null)  (null)
NC_000913.3     2000001 .       T       TC      0.0     PASS            (null)  (null)
NC_000913.3     2000001 .       T       TG      0.0     PASS            (null)  (null)
NC_000913.3     2000001 .       TA      T       0.0     PASS            (null)  (null)
NC_000913.3     2000002 .       A       G       0.0     PASS            (null)  (null)
NC_000913.3     2000002 .       AC      A       0.0     PASS            (null)  (null)
NC_000913.3     2000002 .       A       T       0.0     PASS            (null)  (null)
NC_000913.3     2000002 .       A       C       0.0     PASS            (null)  (null)
NC_000913.3     2000002 .       A       AT      0.0     PASS            (null)  (null)
NC_000913.3     2000002 .       A       AG      0.0     PASS            (null)  (null)
NC_000913.3     2000002 .       A       AC      0.0     PASS            (null)  (null)
NC_000913.3     2000003 .       C       A       0.0     PASS            (null)  (null)
NC_000913.3     2000003 .       C       CA      0.0     PASS            (null)  (null)
NC_000913.3     2000003 .       C       CG      0.0     PASS            (null)  (null)
NC_000913.3     2000003 .       C       CT      0.0     PASS            (null)  (null)
NC_000913.3     2000003 .       C       G       0.0     PASS            (null)  (null)
NC_000913.3     2000003 .       C       T       0.0     PASS            (null)  (null)
NC_000913.3     2000004 .       C       CA      0.0     PASS            (null)  (null)
NC_000913.3     2000004 .       CA      C       0.0     PASS            (null)  (null)
NC_000913.3     2000004 .       C       T       0.0     PASS            (null)  (null)
```

This continues throughout the run - every position is the same.

**Comment by jts** [comment]
It looks like nanopolish isn't finding any usable read data. Maybe there is
a mapping problem? Do you see reads mapped with quality >20 to this region?
Are the read names consistent between the bam and the input fasta/fastq
file?

Jared

**Comment by nickloman** [comment]
Are the reads trimmed? I have seen something like this before with porechop trimmed reads (nanopolish relies on input reads being equal to the equivalent basecalls in the FAST5 file).

**Comment by scottgigante** [comment]
Okay, I've worked it out (mostly):
- There was an unidentified issue with my original data (I did at one point remove blank lines from my fasta file - not sure if this was the culprit or something else) which caused all output to be empty. I verified that every range gave no output
- Once I rebuilt the fasta and bam files, it took a while to reach a section of the genome which was not just masked with Ns, so when I looked at the results of the first few ranges, the output was empty - I didn't bother looking at all of the other output (which was likely fine.)
- I ran on a smaller test sample of E.coli, for which the fast5 files had been moved into numbered subfolders. Since the directory nanopolish was looking in existed, even though the fast5s didn't, no error was thrown. I didn't suspect this error since errors were thrown complaining about fast5 access whenever I ran from a different directory, so I assumed they were indeed available when no error was thrown.

Long story short: dropping to a smaller (old) dataset led me on a wild goose chase, but it would probably be good to throw a warning if no fast5s are found, even if the parent directory is there.

**Comment by jts** [comment]
Thanks for the update. I would have thought it fails if the fast5s aren't found: 

https://github.com/jts/nanopolish/blob/master/src/nanopolish_squiggle_read.cpp#L101

I'll look into why the assertion didn't trip.

**Comment by jts** [comment]
With the overhaul to fast5 loading it is not clear this is still an issue so I'm closing this
